### PR TITLE
Improve docs for add file procedures in Iceberg

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Reword some aspects to make it more precise and ensure that it is clear that the catalog and system schema must be specified.

Feel free to also chime in @findinpath @raunaqmorarka @mayankvadariya @SemionPar and others from the original code PR. 

## Additional context and related issues

Follow up to docs from #22751

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/connector/iceberg.md [hunk 1]
```diff
@@ -568,11 +568,13 @@ nested directories, or `false` to ignore them.
 (iceberg-add-files)=
 #### Add files
 
-The connector can add files from tables or locations if
+The connector can add files from tables or locations to an existing table if
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:571]: Also .. maybe we need to specify .. from Hive tables .. unless you can pull from other table formats as well

File: docs/src/main/sphinx/connector/iceberg.md [hunk 2]
```diff
@@ -584,17 +586,19 @@ relevant schema and table names supplied with the required parameters
 `schema_name` and `table_name`:
 
 ```sql
-ALTER TABLE testdb.iceberg_customer_orders EXECUTE add_files_from_table(
+ALTER TABLE testdb.iceberg_customer_orders 
+EXECUTE example.system.add_files_from_table(
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:590]: The intro sentence talks about the the requirement for the example catalog .. but it is never actually fulfilled.. question would be .. does a use statement before this query work and then it works without example.system ?

Comment (ebyhr) [reply docs/src/main/sphinx/connector/iceberg.md:590]: It works without `example.system` when the default session catalog is `example`.

**Review by ebyhr — APPROVED** [review]

File: docs/src/main/sphinx/connector/iceberg.md (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.md:590]: Added that info.

**Pull request merged**